.NET/Microsoft.Recognizers.Text.DateTime/Models/DateTimeModel.cs

```diff
@@ -55,15 +55,32 @@ public List<ModelResult> Parse(string query, System.DateTime refTime)
                 // No result.
             }
 
-            return parsedDateTimes.Select(o => new ModelResult
+            return parsedDateTimes.Select(o => GetModelResult(o)).ToList();
+        }
+
+        private ModelResult GetModelResult(DateTimeParseResult parsedDateTime)
+        {
+            var modelResult = new ModelResult
             {
-                Start = o.Start.Value,
-                End = o.Start.Value + o.Length.Value - 1,
-                TypeName = o.Type,
-                Resolution = o.Value as SortedDictionary<string, object>,
-                Text = o.Text,
-                ParentText = GetParentText(o)
-            }).ToList();
+                Start = parsedDateTime.Start.Value,
+                End = parsedDateTime.Start.Value + parsedDateTime.Length.Value - 1,
+                TypeName = parsedDateTime.Type,
+                Resolution = parsedDateTime.Value as SortedDictionary<string, object>,
+                Text = parsedDateTime.Text
+            };
+
+            var parentText = GetParentText(parsedDateTime);
+            if (!string.IsNullOrEmpty(parentText))

```

Comment:
This shouldn't be based on the string value. This should be based on the flag for extendedtypes being enabled or not, no?